pybliometrics/sciencedirect/object_metadata.py

```diff
@@ -0,0 +1,62 @@
+"""Module with the ObjectMetadata class."""
+
+from typing import Optional, Union
+
+from pybliometrics.superclasses import Retrieval
+from pybliometrics.utils import chained_get, check_parameter_value, detect_id_type, VIEWS
+
+
+class ObjectMetadata(Retrieval):
+    """Class to retrieve a the metadata of all objects of a document."""
+    @property
+    def results(self) -> list[dict]:
+        """List with metadata of objects in a document. The metadata includes the `url`, `eid`,
+        `ref`, `filename`, `mimetype`, `size`, `height`, `width`, and `type` of the object.
+        """
+        refs = chained_get(self._json, ['attachment-metadata-response', 'attachment'], [])
+        out = []
+        for ref in refs:
+            out.append({'url': ref.get('prism:url'),
+                        'eid': ref.get('eid'),
+                        'ref': ref.get('ref'),
+                        'filename': ref.get('filename'),
+                        'mimetype': ref.get('mimetype'),
+                        'size': ref.get('size'),
+                        'height': ref.get('height'),
+                        'width': ref.get('width'),
+                        'type': ref.get('type')})
```

Comment:
This looks like a list of dicts. Usually, pybliometrics returns a list of namedtuples. Does something speak against using namedtuples here?